app/src/main/java/nu/yona/app/ui/BaseActivity.java

```diff
@@ -34,35 +32,32 @@
 	private InputMethodManager inputMethodManager;
 	private Runnable backPressListener = null;
 	private PauseResumeHook hook;
+	private static int progressDialogCount = 0;
+
+
+	public void toggleLoadingView(boolean loading)
+	{
+		toggleLoadingView(loading, null);
+	}
 
 	/**
 	 * Show loading view.
 	 *
 	 * @param loading the loading
 	 * @param message the message
 	 */
-	public void showLoadingView(boolean loading, String message)
+
+	public void toggleLoadingView(boolean loading, String message)
 	{
 		try
 		{
-			String dialogText = getResources().getString(R.string.loading);
-			if (!TextUtils.isEmpty(message))
-			{
-				dialogText = message;
-			}
-
-       /* if (inputMethodManager == null) {
-            inputMethodManager = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
-        }*/
 			if (loading && progressDialog == null)
 			{
-				progressDialog = new CustomProgressDialog(this, false);
-				progressDialog.show();
+				displayLoadingView();
 			}
 			else if (progressDialog != null && !loading)
 			{
-				progressDialog.dismiss();
-				progressDialog = null;
+				dismissLoadingView();
 			}
 		}
 		catch (Exception e)
```

Comment:
As we're not expecting an exception here, we shouldn't be catching it. Higher layers will take care of it.